actions/debootstrap_action.go

```diff
@@ -80,13 +81,63 @@ func NewDebootstrapAction() *DebootstrapAction {
 	// Be secure by default
 	d.CheckGpg = true
 	// Use main as default component
-	d.Components = []string {"main"}
+	d.Components = []string{"main"}
 	// Set generic default mirror
 	d.Mirror = "http://deb.debian.org/debian"
 
 	return &d
 }
 
+func (d *DebootstrapAction) listOptionFiles(context *debos.DebosContext) []string {
+	files := []string{}
+	if d.Certificate != "" {
+		d.Certificate = debos.CleanPathAt(d.Certificate, context.RecipeDir)
+		files = append(files, d.Certificate)
+	}
+
+	if d.PrivateKey != "" {
+		d.PrivateKey = debos.CleanPathAt(d.PrivateKey, context.RecipeDir)
+		files = append(files, d.PrivateKey)
+	}
+
+	if d.KeyringFile != "" {
+		d.KeyringFile = debos.CleanPathAt(d.KeyringFile, context.RecipeDir)
+		files = append(files, d.KeyringFile)
+	}
+
+	return files
+}
+
+func (d *DebootstrapAction) Verify(context *debos.DebosContext) error {
+	files := d.listOptionFiles(context)
+
+	// Check if all needed files exists
+	for _, f := range files {
+		if _, err := os.Stat(f); os.IsNotExist(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func (d *DebootstrapAction) PreMachine(context *debos.DebosContext, m *fakemachine.Machine, args *[]string) error {
+
+	mounts := d.listOptionFiles(context)
+
+	// Mount configuration files outside of recipes directory
+	for _, mount := range mounts {
+		mount = path.Dir(mount)
+
+		// Check with "/" suffix to avoid false results for the directory
+		// which includes the RecipeDir path in the name, i.e. "/some/dir/RecipeDirExtended"
+		if !strings.HasPrefix(mount+"/", context.RecipeDir+"/") {
+			m.AddVolume(mount)
```

Comment:
why this special-case for the recipe dir?

Reply:
Hm... I thought I described in comment.
Corner case -- let's assume we have `/some/dir/Recipe` with recipes and `/some/dir/RecipeAdditions` with certificates -- in that case `strings.HasPrefix` will return `true` and mount point will be skipped.
I fall into that during the test.

Reply:
well yes obviously; but why are you not just adding a volume for everything rather then special casing in avoiding the recipedir?

Reply:
wanted to avoid the additional mounts like:
```
virtfs-5       811220760 695403072  74540088  91% /home/d4s/go/src/github.com/go-debos/debos
virtfs-7       811220760 695403072  74540088  91% /home/d4s/go/src/github.com/go-debos/debos/keyring
```
but you're right -- looks like unneeded optimisation